pkg/service/azure/volume.go

```diff
@@ -86,10 +86,30 @@ func (a *AzureController) createVolume(ctx context.Context, req *proto.CreateVol
 	if err != nil {
 		return nil, err
 	}
-	return &proto.CreateVolumeResponse{
+
+	ret := &proto.CreateVolumeResponse{
 		ResourceUri: *res.ID,
 		Volumeid:    name,
-	}, nil
+	}
+
+	if req.DeleteSnapshot {
+		sc, err := getSnapshotClient(cred)
+		if err != nil {
+			a.logger.Errorw("failed to get the snapshot client", "error", err)
+			return ret, nil
+
+		}
+
+		err = a.deleteSnapshot(ctx, sc, cred.ResourceGroup, req.Snapshotid)
+		if err != nil {
+			//we will silently log error and return here for now, we dont want to tell the user that volume creation failed in this case.
+			a.logger.Errorw("failed to delete the snapshot", "error", err)
+			return ret, nil
+		}
```

Comment:
Since we are ignoring the error anyway can we do this in a goroutine so the createVolume call doesn't wait for this to finish?

Reply:
yes, we can. Ill update

Reply:
updated